rmw_opensplice_cpp/src/types.cpp

```diff
@@ -149,48 +185,20 @@ print_discovery_logging(
   }
 }
 
-void
-CustomDataReaderListener::add_information(
-  const DDS::SampleInfo & sample_info,
+void CustomDataReaderListener::add_information(
+  const GuidPrefix_t & participant_guid,
+  const GuidPrefix_t & topic_guid,
   const std::string & topic_name,
-  const std::string & type_name,
-  EndPointType end_point_type)
+  const std::string & topic_type,
+  const EndPointType endpoint_type)
 {
-  // store topic name and type name
-  auto & topic_types = topic_names_and_types_[topic_name];
-  topic_types.insert(type_name);
-  // store mapping to instance handle
-  TopicDescriptor topic_descriptor;
-  topic_descriptor.instance_handle = sample_info.instance_handle;
-  topic_descriptor.name = topic_name;
-  topic_descriptor.type = type_name;
-  topic_descriptors_.push_back(topic_descriptor);
-  if (print_discovery_logging_) {
-    print_discovery_logging("+", topic_name, type_name, end_point_type);
-  }
+  topic_cache.AddTopic(participant_guid, topic_guid, topic_name, topic_type);
+  print_discovery_logging("+", topic_name, topic_type, endpoint_type);
 }
 
-void
-CustomDataReaderListener::remove_information(
-  const DDS::SampleInfo & sample_info,
-  EndPointType end_point_type)
+void CustomDataReaderListener::remove_information(const GuidPrefix_t & topic_guid)
 {
-  // find entry by instance handle
-  for (auto it = topic_descriptors_.begin(); it != topic_descriptors_.end(); ++it) {
-    if (it->instance_handle == sample_info.instance_handle) {
-      if (print_discovery_logging_) {
-        print_discovery_logging("-", it->name, it->type, end_point_type);
-      }
-      // remove entries
-      auto & topic_types = topic_names_and_types_[it->name];
-      topic_types.erase(topic_types.find(it->type));
-      if (topic_types.empty()) {
-        topic_names_and_types_.erase(it->name);
-      }
-      topic_descriptors_.erase(it);
-      break;
-    }
-  }
+  topic_cache.RemoveTopic(topic_guid);
```

Comment:
There used to be a conditional call to `print_discovery_logging`, should we add this back in?